controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -130,25 +139,43 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 	r.cephFilesystemSubVolumeGroup.Namespace = r.namespace
 
 	r.cephClientRBDProvisioner = &rookCephv1.CephClient{}
-	r.cephClientRBDProvisioner.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-provisioner", r.storageConsumer.Name)
+	cephClientName := &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "rbd", CephUserType: "provisioner"}
+	r.cephClientRBDProvisioner.Name = generateHashForCephClient(cephClientName)
 	r.cephClientRBDProvisioner.Namespace = r.namespace
+	r.cephClientRBDProvisioner.Annotations["cephUserType"] = fmt.Sprintf("%s-%s-%s", cephClientName.StorageConsumerName, cephClientName.ClaimID, cephClientName.CephUserType)
```

Comment:
1. Let's use a separate annotation for each
2. Use the annotation name prefix we use for all ocs annotations (I believe it is `ocs.openshift.io`, but please verify it)
3. Move the annotation code into the reconciliation section, not the init section, the code here is used to fill only the info that is needed to fetch the resource. The main reason: if the annotation will not be set in the `createOrUpdate` mutate function then they might be overwritten when we load the resource.


```suggestion
	r.cephClientRBDProvisioner.Annotations["ocs.openshift.io.storageconsumer "] = cephClientName.StorageConsumerName
	r.cephClientRBDProvisioner.Annotations["ocs.openshift.io.storageclaim "] = cephClientName.ClaimID
	r.cephClientRBDProvisioner.Annotations["ocs.openshift.io.cephusertype "] = cephClientName.CephUserType
```